### PR TITLE
Implement Chain Agnostic Agenda And Settings

### PR DESCRIPTION
**Pull request by AdrianAndersen**
Oooooboi, this PR turned to be a bit chonkier than I expected. 

So, essentially, this PR redesignes the Agenda and Settings modals. 

For the Agenda:
- Add "Mine Bookinger"
- Add "Planlagte timer" (Now using the session data instead of the config)
- Add better ghost class detection

For the Settings:
- List all Chain Memberships, and add Button + Dialog to add memberships

Other
- Add linking to any class any week, by passing a weekOffset to the link

This PR depends on https://github.com/mathiazom/rezervo/pull/46

### CODE REVIEW

**Comment by AdrianAndersen** [comment]
Resolves https://github.com/mathiazom/rezervo-web/issues/30

**Comment by AdrianAndersen** [comment]
You probably want to checkout this PR and test for yourself, but here is a little sneak peak:
Screenshot 2024-02-26 at 21 40 54
Screenshot 2024-02-26 at 21 40 20
Screenshot 2024-02-26 at 21 39 45
Screenshot 2024-02-26 at 21 38 57

**Comment by AdrianAndersen** [comment]
Robot icon for automatic booking could be cool

File: src/components/configuration/ConfigBar.tsx [hunk 1]
```diff
@@ -54,37 +58,20 @@ function ConfigBar({
 
     return (
         <>
-            {isLoading ? (
```
Comment (mathiazom) [inline src/components/configuration/ConfigBar.tsx:57]: Why has the progress indicator for `isLoading` been removed?

Comment (AdrianAndersen) [reply src/components/configuration/ConfigBar.tsx:57]: I felt like the fact that this loading is always super speedy warrants removing it. It just adds layout shift IMO. But not super strongly opinionated about this

Comment (mathiazom) [reply src/components/configuration/ConfigBar.tsx:57]: Hmm, that is a fair point :+1:

**Pull request merged**